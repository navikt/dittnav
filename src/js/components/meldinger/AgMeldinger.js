import React, { Component } from 'react';
import conf from 'js/Config';

const i18n = {
  'ag.lenkeTekst': '<b>[ag.lenkeTekst locale:lb, variant:null]</b>',
  'ag.melding.flere': 'Du har fått {0} nye meldinger fra arbeidsgiver',
  'ag.melding.en': 'Du har fått en ny melding fra arbeidsgiver',
};

class AgMeldinger extends Component {
  render() {
    const text = i18n['ag.melding.en'];
    return (
      <a data-ga="Dittnav/Varsel/Arbeidsgiver" className="message clickable" href={`${conf.dittNav.SERVICES_URL}${conf.ARBEID_PATH}?goto=${conf.MELDINGER_NAV_PATH}`}>
        <span className="icon ag-icon" aria-label="snakkeboble-ikon" />
        <div className="texts">
          <p>{text}</p>
          <p>{i18n['ag.lenkeTekst']}</p>
        </div>
      </a>
    );
  }
}

export default AgMeldinger;
