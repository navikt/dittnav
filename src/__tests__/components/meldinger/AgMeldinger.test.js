import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import AgMeldinger from 'js/components/meldinger/AgMeldinger'

test('AgMeldinger without messages', () => {
  const component = ReactTestRenderer.create((<AgMeldinger agMessagesCount={0} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('AgMeldinger with 1 message', () => {
  const component = ReactTestRenderer.create((<AgMeldinger agMessagesCount={1} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('AgMeldinger with 6 messages', () => {
  const component = ReactTestRenderer.create((<AgMeldinger agMessagesCount={5} />));
  expect(component.toJSON()).toMatchSnapshot();
});
