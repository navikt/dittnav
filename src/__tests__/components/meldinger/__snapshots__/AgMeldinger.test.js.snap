// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgMeldinger with 1 message 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/Arbeidsgiver"
  href="http://localhost/sbl/nav_security_check?goto=/sbl/as/minside/meldinger/meldingerNAV.do"
>
  <span
    aria-label="snakkeboble-ikon"
    className="icon ag-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har fått en ny melding fra arbeidsgiver
      </span>
    </p>
    <p>
      <span>
        Se melding
      </span>
    </p>
  </div>
</a>
`;

exports[`AgMeldinger with 6 messages 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/Arbeidsgiver"
  href="http://localhost/sbl/nav_security_check?goto=/sbl/as/minside/meldinger/meldingerNAV.do"
>
  <span
    aria-label="snakkeboble-ikon"
    className="icon ag-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har fått 5 nye meldinger fra arbeidsgiver
      </span>
    </p>
    <p>
      <span>
        Oversikt over meldinger
      </span>
    </p>
  </div>
</a>
`;

exports[`AgMeldinger without messages 1`] = `null`;
